framework/src/org/checkerframework/common/fieldtypeinference/FieldTypeInferenceTransfer.java

```diff
@@ -0,0 +1,112 @@
+ * <tt>org.checkerframework.common.fieldtypeinference.FieldTypeInferenceAnnotatedTypeFactory</tt>.
+ *
+ * @author pbsf
+ */
+
+public class FieldTypeInferenceTransfer extends CFTransfer {
+
+    private final Map<Receiver, AnnotatedTypeMirror> assignedPrivateFields;
+    private final FieldTypeInferenceAnnotatedTypeFactory factory;
+
+    public FieldTypeInferenceTransfer(CFAbstractAnalysis<CFValue, CFStore,
+            CFTransfer> analysis) {
+        super(analysis);
+        assert (analysis.getTypeFactory() instanceof FieldTypeInferenceAnnotatedTypeFactory);
+        factory = (FieldTypeInferenceAnnotatedTypeFactory) analysis.getTypeFactory();
+        assignedPrivateFields = factory.getAssignedPrivateFields();
+    }
+
+    @Override
+    public CFStore initialStore(UnderlyingAST underlyingAST,
+            List<LocalVariableNode> parameters) {
+        CFStore initialStore = super.initialStore(underlyingAST, parameters);
+        for (Receiver r : assignedPrivateFields.keySet()) {
+            for (AnnotationMirror am : assignedPrivateFields.get(r).
+                    getAnnotations()) {
+                initialStore.insertValue(r, am);
+            }
+        }
+        return initialStore;
+    }
+
+    @Override
+    public TransferResult<CFValue, CFStore> visitAssignment(AssignmentNode n,
+            TransferInput<CFValue, CFStore> in) {
+        TransferResult<CFValue, CFStore> result = super.visitAssignment(n, in);
+        assert result instanceof RegularTransferResult;
+        Node rhs = n.getExpression();
+        Receiver expr = FlowExpressions.internalReprOf(analysis.getTypeFactory(),
+                n.getTarget());
+
+        // If this is an assignment to a field of 'this' and this field is
+        // private, then populate the map in the ATF.
+        if (!expr.containsUnknown() && expr instanceof FieldAccess &&
+                ((FieldAccess) expr).getField().getModifiers().
+                contains(Modifier.PRIVATE)) {
+            addAssignedField(expr, factory.getAnnotatedType(rhs.getTree()));
+        }
+        return result;
+    }
+
+    /**
+     * Update the ATM mapped by the parameter <tt>r</tt> in the map
+     * <tt>ClassValAnnotatedTypeFactory.analysis.assignedPrivateFields</tt> to be the LUB
+     * of the parameter <tt>atm</tt> and the current ATM mapped by <tt>r</tt>
+     * in this map.
+     * <p>
+     * If <tt>r</tt> is not a key in the map, it is added to the map
+     * mapping to <tt>atm</tt>.
+     * @param r
+     *      A receiver representing a private field.
+     * @param atm
+     *      An ATM assigned to the private field in <tt>r</tt>.
+     */
+    public void addAssignedField(Receiver r, AnnotatedTypeMirror atm) {
+        Map<Receiver, AnnotatedTypeMirror> assignedPrivateFields = factory.getAssignedPrivateFields();
+        if (assignedPrivateFields.containsKey(r)) {
+            Set<? extends AnnotationMirror> lub = analysis.getTypeFactory().
```

Comment:
am I wrong to say that the factory referenced to lines above and the return value of analysis.getTypeFactory are both references to the same object?  If so, should we just use factory?  This might be moot if we move this method.